pkg/internal/server/grpc/snapshotter_secret.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumesnapshot/v1"
+	snapshotutils "github.com/kubernetes-csi/external-snapshotter/v8/pkg/utils"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// getSnapshotterCredentials returns the snapshotter secret associated with VolumeSnapshotClass, if any.
+// The VolumeSnapshotClass is fetched by name if specified, or else the default VolumeSnapshotClass
+// for the CSI driver is used.
+func (s *Server) getSnapshotterCredentials(ctx context.Context, vsi *volSnapshotInfo) (map[string]string, error) {
+	secretRef, err := s.getSnapshotterSecretRef(ctx, vsi)
+	if err != nil {
+		return nil, err
+	}
+
+	if secretRef == nil {
+		return nil, nil
+	}
+
+	// snapshotutils.GetCredentials gets the secrets for the snapshotter CSI calls.
+	return snapshotutils.GetCredentials(s.kubeClient(), secretRef)
```

Comment:
Using the same logic to fetch and assemble the returned map as is used in the external snapshotter sidecar for its calls to the CSI driver.